ibm/resource_ibm_cis_certificate_upload.go

```diff
@@ -0,0 +1,354 @@
+	certsList = append(certsList, *certsItem)
+	priorityOpt := cisClient.NewChangeCertificatePriorityOptions()
+	priorityOpt.SetCertificates(certsList)
+	priorityResponse, err := cisClient.ChangeCertificatePriority(priorityOpt)
+	if err != nil {
+		log.Printf("Change certificate priority failed: %v", priorityResponse)
+		return err
+	}
+	return resourceCISCertificateUploadRead(d, meta)
+}
+func resourceCISCertificateUploadRead(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisSSLClientSession()
+	if err != nil {
+		return err
+	}
+
+	certID, zoneID, crn, err := convertTfToCisThreeVar(d.Id())
+	if err != nil {
+		return err
+	}
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneIdentifier = core.StringPtr(zoneID)
+
+	opt := cisClient.NewGetCustomCertificateOptions(certID)
+	result, response, err := cisClient.GetCustomCertificate(opt)
+	if err != nil {
+		log.Printf("Get custom certificate failed: %v", response)
+		return err
+	}
+	d.Set(cisID, crn)
+	d.Set(cisDomainID, zoneID)
+	d.Set(cisCertificateUploadCustomCertID, result.Result.ID)
+	d.Set(cisCertificateUploadBundleMethod, result.Result.BundleMethod)
+	d.Set(cisCertificateUploadIssuer, result.Result.Issuer)
+	d.Set(cisCertificateUploadHosts, flattenStringList(result.Result.Hosts))
+	d.Set(cisCertificateUploadSignature, result.Result.Signature)
+	d.Set(cisCertificateUploadPriority, result.Result.Priority)
+	d.Set(cisCertificateUploadStatus, result.Result.Status)
+	d.Set(cisCertificateUploadUploadedOn, result.Result.UploadedOn)
+	d.Set(cisCertificateUploadModifiedOn, result.Result.ModifiedOn)
+	d.Set(cisCertificateUploadExpiresOn, result.Result.ExpiresOn)
+	return nil
+}
+func resourceCISCertificateUploadUpdate(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisSSLClientSession()
+	if err != nil {
+		return err
+	}
+
+	certID, zoneID, crn, err := convertTfToCisThreeVar(d.Id())
+	if err != nil {
+		return err
+	}
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneIdentifier = core.StringPtr(zoneID)
+
+	if d.HasChange(cisCertificateUploadBundleMethod) {
+
+		opt := cisClient.NewUpdateCustomCertificateOptions(certID)
+		opt.SetCertificate(d.Get(cisCertificateUploadCertificate).(string))
+		opt.SetPrivateKey(d.Get(cisCertificateUploadPrivateKey).(string))
+		if v, ok := d.GetOk(cisCertificateUploadBundleMethod); ok {
+			opt.SetBundleMethod(v.(string))
+		}
+		_, response, err := cisClient.UpdateCustomCertificate(opt)
+		if err != nil {
+			log.Printf("Update custom certificate failed: %v", response)
+			return err
+		}
+	}
+
+	if d.HasChange(cisCertificateUploadPriority) {
+		// change priority of certificate
+		certsList := []cissslv1.CertPriorityReqCertificatesItem{}
+		id := certID
+		var priority int64
+		if v, ok := d.GetOk(cisCertificateUploadPriority); ok {
+			priority = int64(v.(int))
+		}
+		certsItem, _ := cisClient.NewCertPriorityReqCertificatesItem(id, priority)
```

Comment:
what does this function do?

Reply:
I modified the code such that, if user include the `priority` then it will update the priority for a certificate.  

`cisClient.NewCertPriorityReqCertificatesItem()` this set the payload with `certificate id` and `priority`.